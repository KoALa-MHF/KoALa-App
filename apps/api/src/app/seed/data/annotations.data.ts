import { Annotation } from '../../annotations/entities/annotation.entity';

export const AnnotationsData: Partial<Annotation>[] = [
  {
    id: 1,
    markerId: 1,
    userSessionId: 1,
    start: 200,
  },
];
