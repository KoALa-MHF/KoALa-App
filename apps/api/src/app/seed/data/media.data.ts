import { Media } from '../../media/entities/media.entity';

export const MediaData = [
  {
    id: 1,
    name: 'test.mp3',
    mimeType: 'audio/mpeg',
  },
] as [Media];
