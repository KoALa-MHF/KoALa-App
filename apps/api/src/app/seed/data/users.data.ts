export const UsersData = {
  sessionOwner1: {
    id: 1,
    displayName: 'Session Owner 1',
    email: 'session-owner-1@koala-app.de',
    samlId: '123456',
  },
  sessionOwner2: {
    id: 2,
    displayName: 'Session Owner 2',
    email: 'session-owner-2@koala-app.de',
    samlId: '654321',
  },
  sessionParticipant1: {
    id: 3,
    displayName: 'Session Participant 1',
    email: 'session-participant-1@koala-app.de',
  },
  sessionParticipant2: {
    id: 4,
    displayName: 'Session Participant 2',
    email: 'session-participant-2@koala-app.de',
  },
};
