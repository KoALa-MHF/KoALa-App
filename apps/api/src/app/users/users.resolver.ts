import { Resolver, Mutation, Args, Int, Query } from '@nestjs/graphql';
import { UsersService } from './users.service';
import { User } from './entities/user.entity';
import { UpdateUserInput } from './dto/update-user.input';
import { UseGuards } from '@nestjs/common';
import { AuthGuard } from '../core/guards/auth.guard';
import { CurrentUser } from '../core/decorators/user.decorator';

@Resolver(() => User)
export class UsersResolver {
  constructor(private readonly usersService: UsersService) {}

  @UseGuards(AuthGuard)
  @Query(() => User, { name: 'me' })
  me(@CurrentUser() user: User) {
    return this.usersService.findOne(user.id);
  }

  @UseGuards(AuthGuard)
  @Mutation(() => User)
  updateUser(
    @CurrentUser() user: User,
    @Args('updateUserInput')
    updateUserInput: UpdateUserInput
  ) {
    return this.usersService.update(id, updateUserInput);
  }
}
