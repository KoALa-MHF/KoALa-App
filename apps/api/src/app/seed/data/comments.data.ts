import { UsersData } from './users.data';

export const CommentsData = [
  {
    id: 1,
    owner: UsersData.sessionParticipant1,
    text: 'This is a comment from the Participant',
    annotationId: 1,
  },
  {
    id: 2,
    owner: UsersData.sessionOwner1,
    text: 'This is a comment from the Owner',
    annotationId: 1,
  },
];
