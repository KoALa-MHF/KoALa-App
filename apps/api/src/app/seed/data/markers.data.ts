import { MarkerType } from '../../markers/entities/marker.entity';
import { UsersData } from './users.data';

export const MarkersData = [
  {
    id: 1,
    type: MarkerType.EVENT,
    name: 'Marker 1',
    abbreviation: 'M1',
    description: 'Marker 1 Description',
    color: '#000000',
    icon: 'icon1',
    owner: UsersData.sessionOwner1,
    contentColor: '#FFFFFF',
  },
  {
    id: 2,
    type: MarkerType.RANGE,
    name: 'Marker 2',
    abbreviation: 'M2',
    description: 'Marker 2 Description',
    color: '#000000',
    icon: 'icon2',
    owner: UsersData.sessionOwner1,
    contentColor: '#FFFFFF',
  },
  {
    id: 3,
    type: MarkerType.SLIDER,
    name: 'Marker 3',
    abbreviation: 'M3',
    description: 'Marker 3 Description',
    color: '#000000',
    icon: 'icon3',
    rangeFrom: -5,
    rangeTo: 5,
    owner: UsersData.sessionOwner1,
    contentColor: '#FFFFFF',
  },
  {
    id: 4,
    type: MarkerType.EVENT,
    name: 'Marker 4',
    abbreviation: 'M4',
    description: 'Marker 4 Description',
    color: '#000000',
    icon: 'icon4',
    owner: UsersData.sessionOwner2,
    contentColor: '#FFFFFF',
  },
];
