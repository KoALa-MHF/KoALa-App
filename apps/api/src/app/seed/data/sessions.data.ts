import { SessionStatus } from '../../sessions/entities/session.entity';
import { UsersData } from './users.data';

export const SessionsData = [
  {
    id: 1,
    name: 'Session 1',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.IN_PREPARATION,
  },
  {
    id: 2,
    name: 'Session 2',
    owner: UsersData.sessionOwner2,
    status: SessionStatus.CLOSED,
  },
  {
    id: 3,
    name: 'Session 3',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.IN_PREPARATION,
    start: Date.now(),
    end: Date.now() + 1000000,
  },
  {
    id: 4,
    name: 'Session 4',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.OPEN,
    start: Date.now() - 2000,
    end: Date.now() - 1000,
  },
  {
    id: 5,
    name: 'Session 5',
    owner: UsersData.sessionOwner1,
    status: SessionStatus.ARCHIVED,
  },
];
