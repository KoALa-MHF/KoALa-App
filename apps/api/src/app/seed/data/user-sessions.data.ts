import { UsersData } from './users.data';
import { SessionsData } from './sessions.data';

export const UserSessionsData = [
  {
    owner: UsersData.SESSION_PARTICIPANT_1,
    session: SessionsData[0],
  },
  {
    owner: UsersData.SESSION_PARTICIPANT_2,
    session: SessionsData[0],
  },
  {
    owner: UsersData.SESSION_OWNER_1,
    session: SessionsData[1],
  },
  {
    owner: UsersData.SESSION_PARTICIPANT_2,
    session: SessionsData[2],
  },
];
