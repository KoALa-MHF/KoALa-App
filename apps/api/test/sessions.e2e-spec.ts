import request, { LEGACY_WEBSOCKET_PROTOCOL, supertestWs } from 'supertest-graphql';
import gql from 'graphql-tag';
import { INestApplication } from '@nestjs/common';
import { setupTestApplication } from './mocks/test.util';
import { UsersData } from '../src/app/seed/data/users.data';

const QUERY_SESSIONS = gql`
  query Sessions {
    sessions {
      id
      name
      description
      status
      owner {
        id
        email
        displayName
      }
      toolbars {
        id
      }
      userSessions {
        id
        owner {
          email
        }
        annotations {
          id
          comments {
            id
          }
        }
        session {
          id
          name
        }
      }
    }
  }
`;

const CREATE_SESSION = gql`
  mutation CreateSession($createSessionInput: CreateSessionInput!) {
    createSession(createSessionInput: $createSessionInput) {
      id
      name
      description
      status
      start
      end
      toolbars {
        id
      }
    }
  }
`;

const UPDATE_SESSION = gql`
  mutation UpdateSession($id: Int!, $updateSessionInput: UpdateSessionInput!) {
    updateSession(id: $id, updateSessionInput: $updateSessionInput) {
      id
      name
      description
      status
      start
      end
      editable
      enablePlayer
      displaySampleSolution
      enableLiveAnalysis
      lockAnnotationDelete
      toolbars {
        id
      }
    }
  }
`;

const SESSION_UPDATED = gql`
  subscription onSessionUpdated($id: ID!) {
    sessionUpdated(id: $id) {
      id
      name
      description
      status
      start
      end
      editable
      enablePlayer
      displaySampleSolution
      enableLiveAnalysis
      lockAnnotationDelete
    }
  }
`;

const REMOVE_SESSION = gql`
  mutation RemoveSession($id: Int!) {
    removeSession(id: $id) {
      id
      name
      description
      status
    }
  }
`;

const CREATE_SESSION_VARIABLES = {
  createSessionInput: {
    name: 'test',
  },
};

const UPDATE_SESSION_VARIABLES = {
  id: 1,
  updateSessionInput: {
    name: 'updated name',
    description: 'updated description',
    editable: true,
    enablePlayer: true,
    displaySampleSolution: true,
    enableLiveAnalysis: true,
    lockAnnotationDelete: true,
    start: new Date(),
    end: new Date(),
    status: 'CLOSED',
  },
};

const SESSION_UPDATED_VARIABLES = {
  id: 1,
};

const REMOVE_SESSION_VARIABLES = {
  id: 1,
};

describe('Sessions (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    app = await setupTestApplication();
  });

  afterEach(async () => {
    await app.close();
  });

  describe('Query Sessions', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer()).query(QUERY_SESSIONS);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Owner user should get list of all owned sessions and sessions participating', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_SESSIONS)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });

    it('Participating user should get list of all sessions participating and see only own user sessions and sample solution', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .query(QUERY_SESSIONS)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });
  });

  describe('Create Session', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer()).mutate(CREATE_SESSION).variables(CREATE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('SAML Authenticated user can create a new session', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(CREATE_SESSION)
        .variables(CREATE_SESSION_VARIABLES)
        .expectNoErrors();

      expect(data).toMatchSnapshot();
    });

    it('None SAML Authenticated user cannot create a new session', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(CREATE_SESSION)
        .variables(CREATE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden resource');
    });
  });

  describe('Update Session', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer()).mutate(UPDATE_SESSION).variables(UPDATE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Session Owner can update a session', async () => {
      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(UPDATE_SESSION)
        .variables(UPDATE_SESSION_VARIABLES)
        .expectNoErrors();

      expect(data).toMatchSnapshot({
        updateSession: {
          start: expect.any(String),
          end: expect.any(String),
        },
      });
    });

    it('Session Owner of another session cannot update a session and should get "Not Found" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner2.id}`, { type: 'bearer' })
        .mutate(UPDATE_SESSION)
        .variables(UPDATE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Not Found');
    });

    it('Participant (None SAML Authenticated user) cannot update a session and should get "Forbidden resource" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(UPDATE_SESSION)
        .variables(UPDATE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden resource');
    });

    it('Not authenticated user registering for Session Update Subscription should be denied', async () => {
      // the server needs to listen to an address. This is required by the WS test.
      app.getHttpServer().listen().address();
      try {
        await supertestWs(app.getHttpServer())
          .protocol(LEGACY_WEBSOCKET_PROTOCOL)
          .subscribe(SESSION_UPDATED)
          .variables(SESSION_UPDATED_VARIABLES);

        expect(false).toEqual(true);
      } catch (error) {
        expect(error.message).toEqual('Prohibited connection!');
      }
    });

    it('Session Update triggers SessionUpdated Subscription', async () => {
      // the server needs to listen to an address. This is required by the WS test.
      app.getHttpServer().listen().address();
      const sub = await supertestWs(app.getHttpServer())
        .connectionParams({
          headers: {
            Authorization: `Bearer ${UsersData.sessionOwner1.id}`,
          },
        })
        .protocol(LEGACY_WEBSOCKET_PROTOCOL)
        .subscribe(SESSION_UPDATED)
        .variables(SESSION_UPDATED_VARIABLES);

      await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(UPDATE_SESSION)
        .variables(UPDATE_SESSION_VARIABLES)
        .expectNoErrors();

      const { data } = await sub.next().expectNoErrors();

      expect(data).toMatchSnapshot({
        sessionUpdated: {
          start: expect.any(String),
          end: expect.any(String),
        },
      });
    });
  });

  describe('Remove Session', () => {
    it('Not authenticated user should get "Unauthorized" error', async () => {
      const { errors } = await request(app.getHttpServer()).mutate(REMOVE_SESSION).variables(REMOVE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Unauthorized');
    });

    it('Session Owner can remove a session', async () => {
      const { data: sessionsBeforeDelete } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_SESSIONS)
        .expectNoErrors();
      expect(sessionsBeforeDelete).toMatchSnapshot();

      const { data } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .mutate(REMOVE_SESSION)
        .variables(REMOVE_SESSION_VARIABLES)
        .expectNoErrors();
      expect(data).toMatchSnapshot();

      const { data: sessionsAfterDelete } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner1.id}`, { type: 'bearer' })
        .query(QUERY_SESSIONS)
        .expectNoErrors();
      expect(sessionsAfterDelete).toMatchSnapshot();
    });

    it('Session Owner of another session cannot remove a session and should get "Not Found" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionOwner2.id}`, { type: 'bearer' })
        .mutate(REMOVE_SESSION)
        .variables(REMOVE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Not Found');
    });

    it('Participant (None SAML Authenticated user) cannot remove a session and should get "Forbidden resource" error', async () => {
      const { errors } = await request(app.getHttpServer())
        .auth(`${UsersData.sessionParticipant1.id}`, { type: 'bearer' })
        .mutate(REMOVE_SESSION)
        .variables(REMOVE_SESSION_VARIABLES);

      expect(errors).toHaveLength(1);
      expect(errors[0].message).toBe('Forbidden resource');
    });
  });
});
