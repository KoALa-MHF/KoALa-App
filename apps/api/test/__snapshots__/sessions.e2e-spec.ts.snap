// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions (e2e) Create Session SAML Authenticated user can create a new session 1`] = `
{
  "createSession": {
    "description": "",
    "end": null,
    "id": "4",
    "name": "test",
    "start": null,
    "status": "OPEN",
    "toolbars": [
      {
        "id": "4",
      },
    ],
  },
}
`;

exports[`Sessions (e2e) Query Sessions Owner user should get list of all owned sessions and sessions participating 1`] = `
{
  "sessions": [
    {
      "description": "",
      "id": "1",
      "name": "Session 1",
      "owner": {
        "displayName": "Session Owner 1",
        "email": "session-owner-1@koala-app.de",
        "id": "1",
      },
      "toolbars": [
        {
          "id": "1",
        },
      ],
      "userSessions": [
        {
          "id": 1,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
        {
          "id": 4,
          "owner": {
            "email": "session-participant-1@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
        {
          "id": 5,
          "owner": {
            "email": "session-participant-2@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
      ],
    },
    {
      "description": "",
      "id": "2",
      "name": "Session 2",
      "owner": {
        "displayName": "Session Owner 2",
        "email": "session-owner-2@koala-app.de",
        "id": "2",
      },
      "toolbars": [
        {
          "id": "2",
        },
      ],
      "userSessions": [
        {
          "id": 6,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "2",
            "name": "Session 2",
          },
        },
      ],
    },
    {
      "description": "",
      "id": "3",
      "name": "Session 3",
      "owner": {
        "displayName": "Session Owner 1",
        "email": "session-owner-1@koala-app.de",
        "id": "1",
      },
      "toolbars": [
        {
          "id": "3",
        },
      ],
      "userSessions": [
        {
          "id": 3,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "3",
            "name": "Session 3",
          },
        },
        {
          "id": 7,
          "owner": {
            "email": "session-participant-2@koala-app.de",
          },
          "session": {
            "id": "3",
            "name": "Session 3",
          },
        },
      ],
    },
  ],
}
`;

exports[`Sessions (e2e) Query Sessions Participating user should get list of all sessions participating and see only own user sessions 1`] = `
{
  "sessions": [
    {
      "description": "",
      "id": "1",
      "name": "Session 1",
      "owner": {
        "displayName": "Session Owner 1",
        "email": "session-owner-1@koala-app.de",
        "id": "1",
      },
      "toolbars": [
        {
          "id": "1",
        },
      ],
      "userSessions": [
        {
          "id": 4,
          "owner": {
            "email": "session-participant-1@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
      ],
    },
  ],
}
`;

exports[`Sessions (e2e) Remove Session Session Owner can remove a session 1`] = `
{
  "sessions": [
    {
      "description": "",
      "id": "1",
      "name": "Session 1",
      "owner": {
        "displayName": "Session Owner 1",
        "email": "session-owner-1@koala-app.de",
        "id": "1",
      },
      "toolbars": [
        {
          "id": "1",
        },
      ],
      "userSessions": [
        {
          "id": 1,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
        {
          "id": 4,
          "owner": {
            "email": "session-participant-1@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
        {
          "id": 5,
          "owner": {
            "email": "session-participant-2@koala-app.de",
          },
          "session": {
            "id": "1",
            "name": "Session 1",
          },
        },
      ],
    },
    {
      "description": "",
      "id": "2",
      "name": "Session 2",
      "owner": {
        "displayName": "Session Owner 2",
        "email": "session-owner-2@koala-app.de",
        "id": "2",
      },
      "toolbars": [
        {
          "id": "2",
        },
      ],
      "userSessions": [
        {
          "id": 6,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "2",
            "name": "Session 2",
          },
        },
      ],
    },
    {
      "description": "",
      "id": "3",
      "name": "Session 3",
      "owner": {
        "displayName": "Session Owner 1",
        "email": "session-owner-1@koala-app.de",
        "id": "1",
      },
      "toolbars": [
        {
          "id": "3",
        },
      ],
      "userSessions": [
        {
          "id": 3,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "3",
            "name": "Session 3",
          },
        },
        {
          "id": 7,
          "owner": {
            "email": "session-participant-2@koala-app.de",
          },
          "session": {
            "id": "3",
            "name": "Session 3",
          },
        },
      ],
    },
  ],
}
`;

exports[`Sessions (e2e) Remove Session Session Owner can remove a session 2`] = `
{
  "removeSession": {
    "description": "",
    "id": "1",
    "name": "Session 1",
  },
}
`;

exports[`Sessions (e2e) Remove Session Session Owner can remove a session 3`] = `
{
  "sessions": [
    {
      "description": "",
      "id": "2",
      "name": "Session 2",
      "owner": {
        "displayName": "Session Owner 2",
        "email": "session-owner-2@koala-app.de",
        "id": "2",
      },
      "toolbars": [
        {
          "id": "2",
        },
      ],
      "userSessions": [
        {
          "id": 6,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "2",
            "name": "Session 2",
          },
        },
      ],
    },
    {
      "description": "",
      "id": "3",
      "name": "Session 3",
      "owner": {
        "displayName": "Session Owner 1",
        "email": "session-owner-1@koala-app.de",
        "id": "1",
      },
      "toolbars": [
        {
          "id": "3",
        },
      ],
      "userSessions": [
        {
          "id": 3,
          "owner": {
            "email": "session-owner-1@koala-app.de",
          },
          "session": {
            "id": "3",
            "name": "Session 3",
          },
        },
        {
          "id": 7,
          "owner": {
            "email": "session-participant-2@koala-app.de",
          },
          "session": {
            "id": "3",
            "name": "Session 3",
          },
        },
      ],
    },
  ],
}
`;

exports[`Sessions (e2e) Update Session Session Owner can update a session 1`] = `
{
  "updateSession": {
    "description": "updated description",
    "displaySampleSolution": true,
    "editable": true,
    "enableLiveAnalysis": true,
    "enablePlayer": true,
    "end": Any<String>,
    "id": "1",
    "lockAnnotationDelete": true,
    "name": "updated name",
    "start": Any<String>,
    "status": "CLOSED",
    "toolbars": [
      {
        "id": "1",
      },
    ],
  },
}
`;

exports[`Sessions (e2e) Update Session Session Update triggers SessionUpdated Subscription 1`] = `
{
  "sessionUpdated": {
    "description": "updated description",
    "displaySampleSolution": true,
    "editable": true,
    "enableLiveAnalysis": true,
    "enablePlayer": true,
    "end": Any<String>,
    "id": "1",
    "lockAnnotationDelete": true,
    "name": "updated name",
    "start": Any<String>,
    "status": "CLOSED",
  },
}
`;
